src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -419,23 +419,34 @@ private void putNLPResultToSourceMapForMapType(
         if (sourceValue instanceof Map) {
             for (Map.Entry<String, Object> inputNestedMapEntry : ((Map<String, Object>) sourceValue).entrySet()) {
                 if (sourceAndMetadataMap.get(processorKey) instanceof List) {
-                    // build nlp output for list of nested objects
-                    Iterator<Object> inputNestedMapValueIt = ((List<Object>) inputNestedMapEntry.getValue()).iterator();
-                    for (Map<String, Object> nestedElement : (List<Map<String, Object>>) sourceAndMetadataMap.get(processorKey)) {
-                        // Only fill in when value is not null
-                        if (inputNestedMapValueIt.hasNext() && inputNestedMapValueIt.next() != null) {
-                            nestedElement.put(inputNestedMapEntry.getKey(), results.get(indexWrapper.index++));
+                    if (inputNestedMapEntry.getValue() instanceof List) {
+                        // build nlp output for object in sourceValue which is list type
+                        Iterator<Object> inputNestedMapValueIt = ((List<Object>) inputNestedMapEntry.getValue()).iterator();
+                        for (Map<String, Object> nestedElement : (List<Map<String, Object>>) sourceAndMetadataMap.get(processorKey)) {
+                            // Only fill in when value is not null
+                            if (inputNestedMapValueIt.hasNext() && inputNestedMapValueIt.next() != null) {
+                                nestedElement.put(inputNestedMapEntry.getKey(), results.get(indexWrapper.index++));
+                            }
                         }
+                    } else if (inputNestedMapEntry.getValue() instanceof Map) {
```

Comment:
I would rather have same method name. 
```
if (entryValue instanceof List) { 
  processEntry(entryKey, (List<Object>) entryValue, processorKey, results, indexWrapper, sourceAndMetadataMap); 
} 
else if (entryValue instanceof Map) { 
  processEntry(entryKey, (Map<String, Object>) entryValue, processorKey, results, indexWrapper, sourceAndMetadataMap); 
}


private void processEntry(..., List<Object> entryValue, ...){...}
private void processEntry(..., Map<String, Object> entryValue, ...){...}

```

Reply:
Thanks for sharing the ideas. I may go with @martin-gaievski 's suggestion since I prefer specific method name so that it is more readable.

Reply:
In java, having same method name with different parameter signature is common pattern that is encouraged.

Reply:
Thanks @heemin32 . Given this file already has such pattern of separating methods with name for List/Map type, such as `putNLPResultToSourceMapForMapType()`, `buildNLPResultForListType`. I may keep this pattern. Please feel free to let me know if you have any concerns or ideas. Thank you.

Reply:
1. In `putNLPResultToSourceMapForMapType()`, the map type refers to `sourceValue`, which is an Object. As such, it's explicitly mentioned in the method name. However, I believe a better name would be `putNLPResultToSourceMap()` and that the method should use a more specific type instead of accepting a generic Object.  
2. Consistency with other parts of the code isn't strictly necessary, especially when there's a clearly better approach.  

That said, I won't insist further. Thanks!

Reply:
Cool. That makes sense to me. Thank you @heemin32